### PR TITLE
Allow kubeconfig to be added to the secret

### LINKED ISSUE #1946: kubectl does not use spec.cloudProviders[].config.kubeConfigPath
**What happened**:
When running `kubectl apply`, it seems to read the default kubeconfig and not the kubeconfig configured in `cloudProviders[].config.kubeConfigPath`.

https://github.com/pipe-cd/pipe/blob/2721206ea8b8ed484402ab75dfbafb9b999a8252/pkg/app/piped/cloudprovider/kubernetes/kubectl.go#L27-L31

`Kubectl.config` is defined, but

https://github.com/pipe-cd/pipe/blob/2721206ea8b8ed484402ab75dfbafb9b999a8252/pkg/app/piped/cloudprovider/kubernetes/kubectl.go#L33-L38

the value is `nil`.

https://github.com/pipe-cd/pipe/blob/2721206ea8b8ed484402ab75dfbafb9b999a8252/pkg/app/piped/cloudprovider/kubernetes/kubectl.go#L50-L54

And it's not even referenced when running `kubectl apply`.

**What you expected to happen**:

When applying, the following command will be executed.

```
kubectl apply --kubeconfig=/path/to/cloudproviders/kubeconfig -f -
```

**How to reproduce it**:

Configure the CloudProvider as follows

```
spec:
  cloudProviders:
    - name: kube
      type: KUBERNETES
      config:
        masterURL: https://X.X.X.X:6443
        kubeConfigPath: /etc/kube/kubeconfig
```

And mount kubeconfig to the specified path.

**Environment**:
- `piped` version: v0.9.11
- `control-plane` version: v0.9.11
- Others:

### PR DESCRIPTION
**Pull request by tennashi**
**What this PR does / why we need it**:
Added support for adding kubeconfig to the Secret resource.

There are a few things to consider

* Should we be able to add multiple kubeconfigs?
* Is it appropriate for the name kubeconfig to appear here?
  * Would a name like `general` be better?

**Which issue(s) this PR fixes**:

Fixes #1946
Related to #1960

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: manifests/piped/values.yaml [hunk 1]
```diff
@@ -77,6 +77,7 @@ secret:
   datadogApplicationKey:
     fileName: datadog-application-key
     data: ""
+  kubeConfigs: []
```
Comment (nghialv) [inline manifests/piped/values.yaml:80]: Could you add a comment for this?

```
The list of kubeConfig should be embedded in the Secret to be used by Piped.
Each item of this list has 2 fields "fileName" and "data".
```

**Review by nghialv — COMMENTED** [review]
Thank you. I have just added a nit. Please check again.

File: manifests/piped/values.yaml (hunk 1 above)
Comment (tennashi) [reply manifests/piped/values.yaml:82]: Thanks!
I've added.

**Comment by nghialv** [comment]
Nice. Thank you.
/approve

**Pull request merged**